var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -55,15 +85,64 @@ def cmake_args(self):
                 'ON' if '+aligned' in spec else 'OFF'),
             '-DZFP_WITH_CACHE_TWOWAY:BOOL={0}'.format(
                 'ON' if '+twoway' in spec else 'OFF'),
-            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(
-                'ON' if '+shared' in spec else 'OFF'),
-            '-DBUILD_TESTING:BOOL={0}'.format(
-                'ON' if self.run_tests else 'OFF')
+            '-DZFP_WITH_CACHE_TWOWAY:BOOL={0}'.format(
+                'ON' if '+twoway' in spec else 'OFF')
         ]
-        if spec.version >= Version('0.5.2'):
-            args.append('-DZFP_WITH_CACHE_FAST_HASH:BOOL={0}'.format(
-                'ON' if '+fasthash' in spec else 'OFF'))
-            args.append('-DZFP_WITH_CACHE_PROFILE:BOOL={0}'.format(
-                'ON' if '+profile' in spec else 'OFF'))
+
+        # cfp compressed-array C bindings
+        if '+c' in spec:
+            if not spec.satisfies('@0.5.4:'):
+                raise InstallError('+c requires zfp 0.5.4 or later')
```

Comment:
Rather than raising `InstallError`, which would fail at build time after having installed all the build dependencies, you can use a `conflicts` directive:
```
conflicts('+c', when='@:0.5.3', msg='+c requires zfp 0.5.4 or later')
```
The information from a `conflicts` directive is used at concretization time either to validate a solution and eventually fail with the given error message (original concretizer), or to reduce the search space for a concretized spec (ASP based concretizer).

Reply:
Nice!  Yes, let's go with that.

Reply:
Although if we go with a 'bindings' variant, this would have to be reworked a bit.